Add Delta Lake cookbook recipe (#50)
spiceai/cookbook: delta-lake/README.md

# Delta Lake Data Connector

> Spice can read data straight from a Delta Lake Table. This recipe will create an app that load and query a dataset directly from Delta Lake Tables. It assumes:

Reviewer: ```suggestion
Spice supports reading data directly from Delta Lake tables. This recipe will create an app that loads and queries a dataset from a Delta Lake table in AWS S3. It assumes:
```

- Spice is installed (see the [Getting Started](https://docs.spiceai.org/getting-started) documentation).

> - A Delta Lake Table already exists available in AWS s3.

Reviewer: ```suggestion
- A Delta Lake table is configured and available in AWS S3.
```

- Basic AWS authentication is configured (with environment variable credentials `AWS_ACCESS_KEY_ID` & `AWS_SECRET_ACCESS_KEY`).

1. Initialize a Spice app

   ```shell
   spice init delta_lake_demo
   cd delta_lake_demo
   ```

> 2. Run the following command to set AWS secrets to access Delta Lake table in s3.

Reviewer: ```suggestion
2. Run the following command to set AWS secrets to access the Delta Lake table in S3.
```

   ```bash
   export AWS_ACCESS_KEY_ID=<your-aws-access-key-id>
   export AWS_SECRET_ACCESS_KEY=<your-aws-secret-access-key>
   ```

> 3. Configure the spicepod.yaml as following, replace the `<s3://my_bucket/path/to/s3/delta/table/>` with the actual delta lake table path in S3.

Reviewer: ```suggestion
3. Configure the `spicepod.yaml` as following, replacing the `<s3://my_bucket/path/to/s3/delta/table/>` with the Delta Lake table path in S3.
```

   ```yaml
   version: v1
   kind: Spicepod
   name: delta_lake_demo
   datasets:
     - from: delta_lake:<s3://my_bucket/path/to/s3/delta/table/>
       name: delta_lake_table
       params:
         delta_lake_aws_access_key_id: ${secrets:AWS_ACCESS_KEY_ID}
         delta_lake_aws_secret_access_key: ${secrets:AWS_SECRET_ACCESS_KEY}
   ```

> 4. Start the Spice runtime, and confirm that runtime has register the table `delta_lake_table`

Reviewer: ```suggestion
4. Start the Spice runtime, and the `delta_lake_table` dataset has been registered:
```

   ```shell
   >>> spice run
   2025/01/17 16:30:47 INFO Checking for latest Spice runtime release...
   2025/01/17 16:30:47 INFO Spice.ai runtime starting...
   2025-01-18T00:30:48.557502Z  INFO runtime::init::dataset: Initializing dataset delta_lake_table
   2025-01-18T00:30:48.561170Z  INFO runtime::flight: Spice Runtime Flight listening on 127.0.0.1:50051
   2025-01-18T00:30:48.561514Z  INFO runtime::metrics_server: Spice Runtime Metrics listening on 127.0.0.1:9090
   2025-01-18T00:30:48.569153Z  INFO runtime::http: Spice Runtime HTTP listening on 127.0.0.1:8090
   2025-01-18T00:30:48.574811Z  INFO runtime::opentelemetry: Spice Runtime OpenTelemetry listening on 127.0.0.1:50052
   2025-01-18T00:30:48.758689Z  INFO runtime::init::results_cache: Initialized results cache; max size: 128.00 MiB, item ttl: 1s
   2025-01-18T00:30:49.116731Z  INFO runtime::init::dataset: Dataset delta_lake_table registered (delta_lake:s3:<s3://my_bucket/path/to/s3/delta/table/>), results cache enabled.
   ```

> 5. In another terminal window, run `spice sql`, and check the table `delta_lake_table` exists from the Spice REPL
> 
>    ```shell
>    >>> spice sql
>     Welcome to the Spice.ai SQL REPL! Type 'help' for help.
> 
>     show tables; -- list available tables
>     sql> show tables;
>     +---------------+--------------+------------------+------------+
>     | table_catalog | table_schema | table_name       | table_type |
>     +---------------+--------------+------------------+------------+
>     | spice         | runtime      | task_history     | BASE TABLE |
>     | spice         | runtime      | metrics          | BASE TABLE |
>     | spice         | public       | delta_lake_table | BASE TABLE |
>     +---------------+--------------+------------------+------------+
> 
>     Time: 0.004799292 seconds. 3 rows.
>    ```

Reviewer: ````suggestion
5. In another terminal window, run `spice sql` and check the `delta_lake_table` dataset exists from the Spice REPL:

    ```sql
    show tables;
    ```

   ```shell
   >>> spice sql
    Welcome to the Spice.ai SQL REPL! Type 'help' for help.

    show tables; -- list available tables
    sql> show tables;
    +---------------+--------------+------------------+------------+
    | table_catalog | table_schema | table_name       | table_type |
    +---------------+--------------+------------------+------------+
    | spice         | runtime      | task_history     | BASE TABLE |
    | spice         | runtime      | metrics          | BASE TABLE |
    | spice         | public       | delta_lake_table | BASE TABLE |
    +---------------+--------------+------------------+------------+

    Time: 0.004799292 seconds. 3 rows.
   ```
````

> 6. Query against the Delta Lake table.
> 
>    ```shell

Reviewer: ````suggestion
6. Query against the Delta Lake table.

   ```sql
    select * from delta_lake_table limit 10;
   ```

   ```shell
````
